src/lib/define-dynamic-block.js

```diff
@@ -0,0 +1,90 @@
+// TODO: access `BlockType` and `ArgumentType` without reaching into VM
+// Should we move these into a new extension support module or something?
+import ArgumentType from 'scratch-vm/src/extension-support/argument-type';
+import BlockType from 'scratch-vm/src/extension-support/block-type';
+
+// TODO: grow this until it can fully replace `_convertForScratchBlocks` in the VM runtime
```

Comment:
```suggestion
/**
 * Define a block using extension info which has the ability to dynamically determine (and update) its layout.
 * This functionality is used for extension blocks which can change its shape based on different state information.
 * (E.g. The control_stop block changes its shape based on which menu item is selected.)
 */
// TODO: grow this until it can fully replace `_convertForScratchBlocks` in the VM runtime
```